addons/account/models/account_analytic_line.py

```diff
@@ -87,3 +87,42 @@ def view_header_get(self, view_id, view_type):
                 account=self.env['account.analytic.account'].browse(self.env.context['account_id']).name
             )
         return super().view_header_get(view_id, view_type)
+    
+    def _compute_key(self, analytic_line):
+        key = []
+        if analytic_line.account_id:
+            key.append(str(analytic_line.account_id.id))
+        key.extend(
+            str(analytic_line[field_name].id)
+            for field_name in analytic_line._get_plan_fnames()
+            if field_name.startswith("x_plan") and analytic_line[field_name]
+        )
+        return ",".join(key)
+    
+    def _update_analytic_distribution(self, move_lines):
+        #updating the analytic distribution field in the account.move.line
+        for move_line in move_lines:
+            new_distribution = {}
+            analytic_lines = move_line.analytic_line_ids
+            total_amount = abs(move_line.debit - move_line.credit)
+            for line in analytic_lines:
+                key = self._compute_key(line)
+                new_distribution[key] = (abs(line.amount) / total_amount) * 100
+            
+            move_line.with_context(edited_from_analytic_line=True).write({'analytic_distribution': new_distribution})
+
+    def write(self, vals):
+        #overriding the write function to update the analytic distribution field in the account.move.line
+        res = super().write(vals)
+        affected_move_lines = self.move_line_id
+        self._update_analytic_distribution(affected_move_lines)
```

Comment:
We don't need to always update when writing, only if we are changing the accounts or the amount of the line (i.e. we don't want to recompute everything if we change the label or if we change some unrelated timesheet stuff)